awx/main/models/inventory.py

```diff
@@ -820,9 +820,47 @@ def _get_related_jobs(self):
 
 
 class HostMetric(models.Model):
-    hostname = models.CharField(primary_key=True, max_length=512)
+    hostname = models.CharField(unique=True, max_length=512)
     first_automation = models.DateTimeField(auto_now_add=True, null=False, db_index=True, help_text=_('When the host was first automated against'))
     last_automation = models.DateTimeField(db_index=True, help_text=_('When the host was last automated against'))
+    last_deleted = models.DateTimeField(null=True, db_index=True, help_text=_('When the host was last deleted'))
+    automated_counter = models.IntegerField(default=0, help_text=_('How many times was the host automated'))
+    deleted_counter = models.IntegerField(default=0, help_text=_('How many times was the host deleted'))
+    deleted = models.BooleanField(
+        default=False, help_text=_('Boolean flag saying whether the host is deleted and therefore not counted into the subscription consumption')
+    )
+    used_in_inventories = models.IntegerField(null=True, help_text=_('How many inventories contain this host'))
+
+    objects = models.Manager()
+    active_objects = HostMetricActiveManager()
+
+    def get_absolute_url(self, request=None):
+        return reverse('api:host_metric_detail', kwargs={'pk': self.pk}, request=request)
+
+    def soft_delete(self):
+        if not self.deleted:
+            self.deleted_counter = (self.deleted_counter or 0) + 1
+            self.last_deleted = now()
+            self.deleted = True
+            self.save(update_fields=['deleted', 'deleted_counter', 'last_deleted'])
```

Comment:
If the thing is already deleted, the API gives a somewhat confusing response, which is indistinguishable from a successful deletion. Not sure what to do about that... either increase the deletion counter anyway or give a 400 error?

Reply:
deletion counter means change from "it was deleted" -> "it is deleted", so that's not this case. 

I'm just for the approach like playbooks do - make operation idempotent. 

If it is already deleted, then there is just no action, it's not an error (moreover when awx is designed by "mass deleting illusion" on UI, but APi is called x-times per record - it's pattern used in awx, not invented here)

I don't see a reason, why user needs to distinguish it.